doc/netplan-yaml.md

```diff
@@ -1771,6 +1771,39 @@ WireGuard specific keys:
       > When the `systemd-networkd` backend (v242+) is used, this can
       > also be an absolute path to a file containing the preshared key.
 
+  - **private-key-flags** (sequence of scalars) – since **0.107**
+
+    > Private key flags used by Network Manager. Possible values are:
+    > `agent-owned`, `not-saved` and `not-required`.
+    >
+    > `agent-owned`: a user-session secret agent is responsible for
+    > providing and storing this secret.
+    >
+    > `not-saved`: this secret should not be saved but should be
+    > requested from the user each time it is required.
+    >
+    > `not-required`: this flag hints that the secret is not required
+    > and should not be requested from the user.
+
+    Example:
+
+    ```yaml
+    network:
+      renderer: NetworkManager
+      tunnels:
+        wg0:
+          mode: wireguard
+          port: 5182
+          private-key-flags:
+            - agent-owned
```

Comment:
Regarding new schema:
This seems to be a rather NetworkManager specific settings, I feel like it should therefore live in the `nm_backend_settings_handlers` instead. Also, I wonder if we could/should make this a bit more generic, by introducing a generic "networkmanager.flags" setting, which might contain other (future) flags, then using the "validation" stage to verify those flags make sense for the given interface definition implicitly (c.f. `wakeonwlan` setting). E.g.:

   ```yaml
    network:
      renderer: NetworkManager
      tunnels:
        wg0:
          mode: wireguard
          networkmanager:
            flags: [private-key-agent-owned]
          port: 5182
          peers:
            - keys:
                public: rlbInAj0qV69CysWPQY7KEBnKxpYCpaWqOs/dLevdWc=
              allowed-ips: [0.0.0.0/0, "2001:fe:ad:de:ad:be:ef:1/24"]
              keepalive: 23
              endpoint: 1.2.3.4:5
      ```

What do you think?

Reply:
That works for me. It could be moved to `backend_settings`, but it's used in the middle of the netdef struct, so adding a new variable there will change the struct layout. Is that fine?

Reply:
On a second thought, I think having a generic `flags` key might be tricky to maintain. We'll have to have our own encoding. What if a flag for a given context have the same value as another flag from a different context in Network Manager?

I think it's fine to move the private-key-flags to the backend specific data though.